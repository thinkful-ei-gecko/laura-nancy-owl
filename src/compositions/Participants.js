import React from 'react';
import './ParticipantsStyle.css';



function Participants(props) {

    const activeParticipants = props.participants.filter(participant => participant.inSession).map(
        participant => (
                <li key={participant.id}>
                    {participant.name}
                    <img src={participant.avatar} alt="{participant.name}"/>
                    <span>{participant.onStage ? 'on stage' : 'observer'}</span>
                </li>))

    return (
        <section className='participantsList'>
            <div className='info'>
                <ul>
                    {activeParticipants}
                </ul>
            </div>
        </section>
    )
}

export default Participants;