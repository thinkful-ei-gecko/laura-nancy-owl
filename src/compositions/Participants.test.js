import React from 'react';
import ReactDOM from 'react-dom';
import Participants from './Participants';
import participants from './ParticipantsList';
import renderer from 'react-test-renderer';

describe('<Participants />', () => {
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<Participants participants = {participants}/>, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders this UI as expected', () => {
        // Render the component, as JSON
        const tree = renderer.create(<Participants participants = {participants}/>).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});