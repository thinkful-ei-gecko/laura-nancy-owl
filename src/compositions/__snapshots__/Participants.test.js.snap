// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Participants /> renders this UI as expected 1`] = `
<section
  className="participantsList"
>
  <div
    className="info"
  >
    <ul>
      <li>
        Koren Templeton
        <img
          alt="{participant.name}"
          src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        />
        <span>
          on stage
        </span>
      </li>
      <li>
        Caty Flucker
        <img
          alt="{participant.name}"
          src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
        />
        <span>
          observer
        </span>
      </li>
      <li>
        Frank Runciman
        <img
          alt="{participant.name}"
          src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
        />
        <span>
          observer
        </span>
      </li>
      <li>
        Ashla Attwool
        <img
          alt="{participant.name}"
          src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
        />
        <span>
          on stage
        </span>
      </li>
    </ul>
  </div>
</section>
`;
